include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -379,10 +375,48 @@ namespace Utilities
 #  endif
         }
 
-        // unpack data
+        // We know from the various calls to MPI_Iprobe that all of our
+        // requests have received answers, but we have not actually
+        // gotten the data from MPI. Do so and unpack the data.
         {
           for (unsigned int i = 0; i < targets.size(); ++i)
-            this->process.read_answer(targets[i], recv_buffers[i]);
+            {
+              const int tag_deliver = Utilities::MPI::internal::Tags::
+                consensus_algorithm_nbx_process_deliver;
+
+#  ifdef DEBUG
+              // Just to be sure, double check that the message really
+              // is already here -- in other words, that the following
+              // MPI_Recv is going to return immediately
+              {
+                int        request_is_pending;
+                const auto ierr = MPI_Iprobe(targets[i],
```

Comment:
Probe?

Reply:
No. I'm interested in asserting that the message has really already arrived. `Iprobe` returns immediately and tells us whether the message is there already; `Probe` hangs until the message is in fact there.